### PR TITLE
hyperpotamus: init at 0.38.2

### PR DESCRIPTION
**Pull request by onny**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/hyperpotamus/node-packages.nix [hunk 1]
```diff
@@ -0,0 +1,1889 @@
...
+    meta = {
+      description = "YAML based HTTP script processing engine";
+      homepage = https://github.com/pmarkert/hyperpotamus/wiki;
+      license = "MIT";
+    };
+    production = true;
+    bypassCache = true;
+    reconstructLock = true;
+  };
+}
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/hyperpotamus/node-packages.nix:1889]: Missing final newline.

File: pkgs/tools/networking/hyperpotamus/node-packages.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+[
+  {"hyperpotamus": "0.38.2"}
+]
+
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/hyperpotamus/node-packages.json:4]: One final new line to much.

File: pkgs/tools/networking/hyperpotamus/default.nix [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  version = "0.38.2";
+in
+(import ./hyperpotamus.nix {
+  inherit pkgs;
+  inherit (stdenv.hostPlatform) system;
+})."hyperpotamus-${version}".override {
+  meta = {
+    description = "YAML/JSON automation scripting";
+    homepage = "https://github.com/pmarkert/hyperpotamus";
+    license = lib.licenses.mit;
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/hyperpotamus/default.nix:12]:
```suggestion
    license = licenses.mit;
```

File: pkgs/tools/networking/hyperpotamus/default.nix [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{ lib, stdenv, pkgs }:
+let
+  version = "0.38.2";
+in
+(import ./hyperpotamus.nix {
+  inherit pkgs;
+  inherit (stdenv.hostPlatform) system;
+})."hyperpotamus-${version}".override {
+  meta = {
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/hyperpotamus/default.nix:9]:
```suggestion
  meta = with lib; {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Why is this not using the global nodePackages?

**Comment by onny** [comment]
> Why is this not using the global nodePackages?

Good point, was unsure about nodejs-packaging. Hopefully fixed it now :)

**Comment by SuperSandro2000** [comment]
I meant why it isn't part of https://github.com/NixOS/nixpkgs/tree/master/pkgs/development/node-packages ?

**Comment by onny** [comment]
> I meant why it isn't part of https://github.com/NixOS/nixpkgs/tree/master/pkgs/development/node-packages ?

Ah now I get it. It's now included in nodePackages!

**Comment by SuperSandro2000** [comment]
@onny and again a merge conflict...

**Comment by onny** [comment]
Hope everything is fine now :)

**Comment by onny** [comment]
Had to create a new PR because I lost access to the original repository :( https://github.com/NixOS/nixpkgs/pull/132813

**Pull request closed without merging**